dali/kernels/imgproc/convolution/convolution_gpu.h

```diff
@@ -236,17 +246,45 @@ struct ConvolutionGpu {
                 "Selected axis must be in [0, ndim) when there is no channel axis, or in [0, ndim "
                 "- 1) for channel-last input");
 
-  void FillAlignedWindows(span<W> window_tmp_buffer_host,
-                         const TensorListView<StorageCPU, const W, 1>& window,
+
+  void FillAlignedWindow(span<W> window_tmp_buffer_host, int window_idx, span<const W> window_src,
                          const TensorListShape<ndim>& in_shape) {
+    using dst_win_t = typename CutlassConv::ConvWindowConfiguration::template PaddedWindowBuffer<W>;
+    int num_channels = has_channels ? in_shape[window_idx][ndim - 1] : 1;
+    auto window_padded_dst = dst_win_t(&window_tmp_buffer_host[window_idx * kWindowCopyBufferSize]);
+    CutlassConv::ConvWindowConfiguration::prepare_window(window_padded_dst, window_src,
+                                                         num_channels);
+  }
+
+  /**
+   * @brief Repack kernel windows from packed TLV representation to a padded layout
+   * suitablbe for CUTLASS kernel taking into account the shape.
+   *
+   * For axis of extent 1 the kernel window is compacted to 1 element.
+   */
+  void FillAlignedWindows(span<W> window_tmp_buffer_host,
+                          const TensorListView<StorageCPU, const W, 1>& window,
+                          const TensorListShape<ndim>& in_shape) {
     for (int i = 0; i < window.num_samples(); i++) {
-      using dst_win_t =
-          typename CutlassConv::ConvWindowConfiguration::template PaddedWindowBuffer<W>;
-      int num_channels = has_channels ? in_shape[i][ndim - 1] : 1;
-      auto window_src = make_span(window.tensor_data(i), window.tensor_shape_span(i)[0]);
-      auto window_padded_dst = dst_win_t(&window_tmp_buffer_host[i * kWindowCopyBufferSize]);
-      CutlassConv::ConvWindowConfiguration::prepare_window(window_padded_dst, window_src,
-                                                           num_channels);
+      // Special case, for axis with extent 1 - we need to sum the window to 1 element,
+      // to make it work with border reflect 101. For the CPU version, the reflect 101
+      // works as border repeat - so with 1 elements it works by calculating
+      // (sum(window) * input_element). The kernel matrix generation ends as infinite loop
+      // for input of extent equal 1, so we compact the kernel to 1 element (what would effectively
+      // happen either way) and just lookup that one elemnt in the CUTLASS kernels
+      if (in_shape[i][axis] == 1) {
+        std::array<W, 1> tmp_window = {W{0}};
+        for (int win_elem = 0; win_elem < window.tensor_shape_span(i)[0] / 2; win_elem++) {
+          tmp_window[0] += window.tensor_data(i)[win_elem];
+          tmp_window[0] += window.tensor_data(i)[window.tensor_shape_span(i)[0] - win_elem - 1];
+        }
+        tmp_window[0] += window.tensor_data(i)[window.tensor_shape_span(i)[0] / 2];
```

Comment:
Why not just a plain loop?
```suggestion
        for (int win_elem = 0; win_elem < window.tensor_shape_span(i)[0]; win_elem++) {
          tmp_window[0] += window.tensor_data(i)[win_elem];
        }
```

Reply:
Use Kahan's sum instead?

Reply:
done